dpm/api/env.py

```diff
@@ -15,38 +19,48 @@ class Env:
     dpm_initialized: bool = False
     secrets_initialized: bool = False
 
-    @staticmethod
-    def initialize(dpm_service_name: str, dpm_program_name: str, secrets_name: str = None, project: str = None,
-                   secrets_polling_interval: int = 0):
-
+    def __init__(
+        self,
+        dpm_service_name: str,
+        dpm_program_name: str,
+        secrets_name: str = None,
+        project: str = None,
+        secrets_polling_interval: int = 0,
+    ):
         # Setting up dynamic properties
-        Env.dpm_service_name = dpm_service_name
-        Env.dpm_program_name = dpm_program_name
-        Env.dpm_client = clients.DynamicPropertyManagementClient(
-            service_name=dpm_service_name, program_name=dpm_program_name)
-        Env.dpm_initialized = True
+        self.dpm_service_name = dpm_service_name
+        self.dpm_program_name = dpm_program_name
+        self.dpm_client = clients.DynamicPropertyManagementClient(
```

Comment:
I'm not sure yet if we'd want this to make a connection call to the clients (DynamicPropertyManagementClient + secrets) each time we call the constructor, if it's already established the same connection... and used in the same module.

Reply:
A library or application constructs a single `env` based upon the constructor then holds a pointer to that in memory.  This would mean that any time I refer to what it points too, that pointer only has 1 instantiation of `dpm_client`.  If you create the object in memory then it goes out of memory by calling the [destructor](https://pythonprogramminglanguage.com/destructor/), the `dpm_client` will get disposed and the connection deleted.  If the caller is wanting to do a singleton pattern, they may if it is fitting to their use case still.

Reply:
The Singleton pattern was what I was intended to get at. The destructor and garbage collection of the object was less of my concern then if we were to create new connections to Firestore each time; I think in the other PR you suggested an alternate approach as well, to push the fields to the env.

Reply:
We definitely don't want to close/re-open connections to Firestore, but if `env` is only instantiated once, we should be okay (but then you have to pass `env` around everywhere?). 
Or we cache the connection to firestore in the class or however one does that

Reply:
I was thinking one of 2 things.  You could pass around the environment as you mentioned or your consuming application could make use of it in some singleton/staticmethod pattern for callers outside of the library.  The former could be passing it around or just doing inversion of control to make this transparent to the caller.